src/Client/ResourceClient.php

```diff
@@ -160,30 +154,29 @@ public function upsertResource($uri, array $uriParameters = [], array $body = []
      */
     public function upsertResourceList($uri, array $uriParameters = [], $resources = [])
     {
-        if (!is_array($resources) && !$resources instanceof \Traversable) {
-            throw new \InvalidArgumentException('The parameter "resources" must be an array or Traversable.');
-        }
-
-        $streamFunction = function () use ($resources) {
+        if (is_array($resources)) {
+            $body = '';
             $isFirstLine = true;
             foreach ($resources as $resource) {
                 if (!is_array($resource)) {
                     throw new \InvalidArgumentException('The parameter "resources" must be an array of array.');
                 }
                 unset($resource['_links']);
-                yield ($isFirstLine ? '' : PHP_EOL) . json_encode($resource);
+                $body .= ($isFirstLine ? '' : PHP_EOL) . json_encode($resource);
                 $isFirstLine = false;
             }
-        };
-
-        $streamedBody = $this->streamFactory->createStream($streamFunction());
+        } elseif ($resources instanceof StreamInterface) {
+            $body = $resources;
+        } else {
```

Comment:
imho, it was more readable with the exception thrown at the beginning of the test